front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -220,3 +220,224 @@ AgentConfiguration.hasOne(AgentRetrievalConfiguration, {
   foreignKey: { name: "agentId", allowNull: true }, // null = no generation set for this Agent
   onDelete: "CASCADE",
 });
+
+/**
+ * Retrieval Action
+ */
+export class AgentRetrievalAction extends Model<
+  InferAttributes<AgentRetrievalAction>,
+  InferCreationAttributes<AgentRetrievalAction>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare query: string | null;
+  declare relativeTimeFrameDuration: number | null;
+  declare relativeTimeFrameUnit: TimeframeUnit | null;
+  declare topK: number;
+
+  declare retrievalConfigurationId: ForeignKey<
+    AgentRetrievalConfiguration["id"]
+  >;
+}
+AgentRetrievalAction.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    query: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+    relativeTimeFrameDuration: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    relativeTimeFrameUnit: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    topK: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "agent_retrieval_action",
+    sequelize: front_sequelize,
+    hooks: {
+      beforeValidate: (retrieval: AgentRetrievalAction) => {
+        // Validation for Timeframe
+        if (
+          retrieval.relativeTimeFrameDuration === null &&
```

Comment:
On my last PR @fontanierh told me that empty fields are not set to `null` but to `undefined`, so this will never be true, and should be replaced by !retrieval.relativeTimeFrameDuration 🙏🏻

Reply:
Roger that

Reply:
What about 0 ?